### PR TITLE
Improve markdown italicizing when following tags

### PR DESCRIPTION
**Pull request by skyweb331**
### Fixed Issues
$ https://github.com/Expensify/App/issues/41110
This PR is to fix italicizing the context which are following `<code>,<pre>,<a>` tags in one line.

# Tests
1. What unit/integration tests cover your change? What autoQA tests cover your change?
Passed all existing test cases and Added one more test case for this purpose.
1. What tests did you perform that validates your changed worked?
bump expensify-common version in E/App and `react-native-live-markdown` to this PR and tested locally
330365055-4134416a-eb49-42fc-8884-697b7f265cee

# QA
1. What does QA need to do to validate your changes?
bump expensify-common version in E/App and use chat normally - sent messages should be parsed with the new rule
1. What areas do they need to test for regressions?
Italic markdown following `<code>,<pre>,<a>,<mention-user>` tags in one line

### CODE REVIEW

**Comment by skyweb331** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by thienlnam** [comment]
Additionally, the GH_LINK text should be replaced with the link https://github.com/Expensify/App/issues/41110 so it would be $ https://github.com/Expensify/App/issues/41110

**Comment by thienlnam** [comment]
@sobitneupane I can't seem to assign you here, so you might have to comment on the issue first

**Comment by skyweb331** [comment]
> Additionally, the GH_LINK text should be replaced with the link [Expensify/App#41110](https://github.com/Expensify/App/issues/41110) so it would be $ [Expensify/App#41110](https://github.com/Expensify/App/issues/41110)

DONE

**Comment by sobitneupane** [comment]
Sorry for the delay. I will review the PR asap.

File: lib/ExpensiMark.js [hunk 1]
```diff
@@ -284,22 +284,21 @@ export default class ExpensiMark {
...
             {
                 /**
-                 * Use \b in this case because it will match on words, letters,
-                 * and _: https://www.rexegg.com/regex-boundaries.html#wordboundary
-                 * The !_blank is to prevent the `target="_blank">` section of the
-                 * link replacement from being captured Additionally, something like
-                 * `\b\_([^<>]*?)\_\b` doesn't work because it won't replace
-                 * `_https://www.test.com_`
-                 * Use [\s\S]* instead of .* to match newline
+                 * Ignore <pre>, <code>, <a>, <mentin-user> tag contents first, so _blank is ignored due to it a property of <a>,
```
Comment (sobitneupane) [inline lib/ExpensiMark.js:287]: @skyweb331 Can you please only remove those comments that are irrelevant after the change?

Comment (skyweb331) [reply lib/ExpensiMark.js]: @sobitneupane Removed the comment

Comment (sobitneupane) [reply lib/ExpensiMark.js]: @skyweb331 Sorry for the misunderstanding. I mean to say please keep older comments as well if they are still relevant.

File: lib/ExpensiMark.js [hunk 2]
```diff
@@ -177,7 +177,7 @@ export default class ExpensiMark {
             {
                 name: 'reportMentions',
 
-                regex: /(?<![^ \n*~_])(#[\p{Ll}0-9-]{1,80})/gmiu,
+                regex: /(?<![^ \n*~_])(#[\p{Ll}0-9-]{1,80})/gimu,
```
[the remark this answers is not in the document]
Comment (skyweb331) [reply lib/ExpensiMark.js:180]: @sobitneupane This is done by my vscode prettier... seems like no meaningful change.

**Review by thienlnam — APPROVED** [review]

**Comment by thienlnam** [comment]
Hmm, looks like we have some commits that are not signed - @skyweb331 could you fix them with steps from https://expensify.slack.com/archives/C01GTK53T8Q/p1711655905519379?thread_ts=1711571483.669109&cid=C01GTK53T8Q?

**Comment by skyweb331** [comment]
@thienlnam I thought commit verification is only for NewDot. I fixed it. Thank you.

**Review by thienlnam — APPROVED** [review]
Thanks!

**Pull request merged**